internal/trie/node/header.go

```diff
@@ -70,24 +81,32 @@ var (
 	ErrPartialKeyTooBig = errors.New("partial key length cannot be larger than 2^16")
 )
 
-func decodeHeader(reader io.Reader) (variant byte,
+func decodeHeader(reader io.Reader) (nodeVariant variant,
 	partialKeyLength uint16, err error) {
```

Comment:
Hey Eclesio, good catch! I didn't revisited that code in detail but I think it could be related [with this](https://github.com/paritytech/substrate/blob/dfd82860feb5da6f74ca2aa04414d9cebe310204/primitives/trie/src/node_header.rs#L113-L114)